src/app.rs

```diff
@@ -491,3 +502,24 @@ lazy_static! {
 fn validate_number(s: String) -> Result<(), String> {
     s.parse::<usize>().map(|_|()).map_err(|err| err.to_string())
 }
+
+fn validate_max_filesize(s: String) -> Result<(), String> {
+    let re = Regex::new(r#"^(\d+)([KMG])?$"#).unwrap();
+    let caps = try!(re.captures(&s)
+                      .ok_or("invalid format for max-filesize argument"));
+
+    let value = caps.get(1);
+    let suffix = caps.get(2).map(|x| x.as_str());
+
+    match value {
```

Comment:
You don't need this match. You can unwrap the `value` since it is guaranteed to match if the entire regex matched. e.g., `try!(caps[1].parse::<u64>().map_err(|err| err.to_string()))`.